server/routes/api/series.py

```diff
@@ -58,6 +59,58 @@ def series_within_core(parent_entity, child_type, variables, all_facets):
   return compact_series(resp, all_facets)
 
 
+# TODO(juliawu): Handle case where dates are not "YYYY-MM"
+# TODO(juliawu): We should adjust how we bin based on the data,
+#                instead of cutting it off.
+def get_binned_series(entities, variables, year):
+  """Get observation series for entities and variables, for a given year.
+  
+  Bins observations from a series for plotting in the histogram tile.
+  Currently, binning is done by only returning observations for a given year.
+  This is done by assuming the dates of the series are in 'YYYY-MM' format,
+  filtering for observations from the given year, and then imputing 0 for
+  any months missing, up to the end of the series. 
+
+  Note: This assumes the dates of the series are in 'YYYY-MM' format.
+
+  Args:
+    entities (list): DCIDs of entities to query
+    variables (list): DCIDs of variables to query
+    year (str): year in "YYYY" format to get observations for
+  
+  Returns:
+    JSON response from server, with series containing only observations from 
+    the specified year.
+  """
+  # Get raw series from mixer
+  data = series_core(entities, variables, False)
+
+  for stat_var in variables:
+    for location in data['data'][stat_var].keys():
+
+      # filter series to just observations from the selected year
+      series = data['data'][stat_var][location]['series']
```

Comment:
nit: use `.get` with defaults for most of these dereferences to be resilient to server errors or missing data.